### ISSUE TITLE
Line Numbers in CLI Sample only does line 0

### ISSUE THREAD

**Issue by kwalkerxxi** [Swiftly1/CsharpToColouredHTML#12]
Firstly, Just found this a few days ago and think this is a great library. I need something like this to embed nicely formatted code as HTML as the CMS I use is limited in what can be displayed within it.

Issue: I ran the CLI (using your sample test) and added line numbers but it only does line 0. See Attached Sample Output and slightly modified version of the CLI Program.cs. Any tips or suggestions on what I might be doing incorrectly?

![2022-11-16_22-58-06](https://user-images.githubusercontent.com/4986692/202187444-9ce5d43c-2bad-474c-891f-c46f099c97d9.png)
![2022-11-16_22-56-55](https://user-images.githubusercontent.com/4986692/202187470-b7500a67-9ef9-45d7-bfac-fbfb4ceaeee1.png)

**Comment by Swiftly1** [comment]
@kwalkerxxi 

Hi,

I'm pretty sure it is about line endings

I tend to use `Environment.NewLine`, so if the operating system you use has different line endings than the text you're using,
then the result will be messy - everything will be treated as one line.

So, the quick solution is to have line endings consistent with your OS.

For reference - `Environment.NewLine` behaves like this:

>\r\n for non-Unix platforms, or \n for Unix platforms.

Please let me know if it solved your issue.

**Comment by Swiftly1** [comment]
On this branch https://github.com/Swiftly1/CsharpToColouredHTML/tree/configurable-newline-endings

I've pushed some experimental solution that should fix your issue

```csharp
var colourer_settings = new ColourerSettings
{
    ConvertNewLineEndingsToEnvironmentNewLine = true
};

var html = new CsharpColourer(colourer_settings).ProcessSourceCode(code, new HTMLEmitter());
```

If you could give me confirmation whether it solved your issue, then that would be nice

FYI: In the final version that will be merged that API may be a little bit different (e.g property name)
or maybe I will enable it by default, so no configuration will be needed at all, but I'll think about it once I know that it works for you

**Comment by Swiftly1** [comment]
@kwalkerxxi 

Hi again, 

I've published version `1.0.34` which should fix that issue by normalizing new line endings used in input code to Environment.NewLine.

No changes to code required. 

Please let me know if it solved your issue

**Comment by Swiftly1** [comment]
Seems like this issue has been solved since there's no response from @kwalkerxxi 

I'm closing this, feel free to open if there's something wrong

**Comment by kwalkerxxi** [comment]
@Swiftly1 Sorry about the slow response on this one. I feel really bad about not replying - my apologies. I got busy at work and wasn't as engaged with tech over the Christmas break. I can confirm things seem to be outputting well. Thanks for the fix!

I have now started using your great library for outputting a mass of C Sharp files into HTML. I jumped back on to post an issue with generic methods not formatting well and noticed you have improved that too! Thanks for that as well. 

I'll try to more actively watch the issue requests I post here in the future.
